packages/dart/npt_flutter/lib/features/onboarding/widgets/onboarding_button.dart

```diff
@@ -204,51 +220,59 @@ class _OnboardingButtonState extends State<OnboardingButton> {
-        App.navState.currentContext!.read<BackupKeyCubit>().setBackupKeyStatus(false);
-        final apiKey = await Constants.appAPIKey;
+        var context = App.navState.currentContext!;
+        // ignore: use_build_context_synchronously
+        context.read<BackupKeyCubit>().setBackupKeyStatus(false);
+        // ignore: use_build_context_synchronously
+        final rootDomain = context
+            .read<OnboardingCubit>()
+            .getRootDomain(); // Or get from your state/cubit if needed
+        final root =
+            // ignore: use_build_context_synchronously
+            Constants.getRootDomains(context)[rootDomain];
 
-        if (apiKey == null) {
-          result = AtOnboardingResult.error(
-            message: strings.errorAtSignNotExist,
-          );
-          break;
+        final apiKey = await root?.apiKey;
+        if (apiKey != null) {
+          AtOnboardingConstants.setApiKey(apiKey!);
         }
-        AtOnboardingConstants.setApiKey(apiKey);
-        AtOnboardingConstants.rootDomain = util.config.atClientPreference.rootDomain;
+        AtOnboardingConstants.rootDomain =
+            util.config.atClientPreference.rootDomain;
 
-        await AtOnboardingLocalizations.load(LanguageUtil.getLanguageFromLocale(Locale(Platform.localeName)).locale);
+        await AtOnboardingLocalizations.load(
+            LanguageUtil.getLanguageFromLocale(Locale(Platform.localeName))
+                .locale);
         if (!mounted) return null;
-        Map<String, String> apis = {
-          "root.atsign.org": "my.atsign.com",
-          "root.atsign.wtf": "my.atsign.wtf",
-        };
-        var regUrl = apis[util.config.atClientPreference.rootDomain];
-        if (regUrl == null) {
-          result ??= AtOnboardingResult.error(
-            message: strings.errorRootDomainNotSupported,
-          );
-          break;
-        }
+
         result = await showDialog<AtOnboardingResult>(
+          // ignore: use_build_context_synchronously
           context: context,
           barrierDismissible: false,
-          builder: (context) => ActivateAtsignDialog(
-            atSign: atsign,
-            apiKey: apiKey,
-            config: util.config,
-            registrarUrl: regUrl,
-            onboardingUtil: util,
-            waitForTeapot: initialStatus != AtSignStatus.teapot,
-          ),
+          builder: (context) => (root?.registrarUrl == null || apiKey == null)
+              ? ActivateCramDialog(
+                  atSign: atsign,
+                  config: util.config,
+                  onboardingUtil: util,
+                )
+              : ActivateOtpDialog(
+                  atSign: atsign,
+                  apiKey: apiKey,
+                  config: util.config,
+                  registrarUrl: root!.registrarUrl!,
+                  onboardingUtil: util,
+                  waitForTeapot: initialStatus != AtSignStatus.teapot,
+                ),
```

Comment:
Here's where we pick the activation screen based on whether apikey & reg url are known or not.